lib/scraper_test.rb

```diff
@@ -30,17 +31,42 @@ def install_tasks
           it 'should contain the expected data' do
             # TODO: Make this configurable
             Dir['test/data/*.yml'].each do |file|
-              yaml_data = YAML.load_file(file).to_h
-              url = yaml_data[:url]
-              class_to_test = Object.const_get(yaml_data[:class])
-              VCR.use_cassette(File.basename(url)) do
-                response = class_to_test.new(response: Scraped::Request.new(url: url).response)
-                response.to_h.must_equal yaml_data[:to_h]
+              VCR.use_cassette(File.basename(file)) do
+                instructions = Instructions.new(FilePath.new(file))
+                response = instructions.class_to_test
+                                       .new(response: Scraped::Request.new(url: instructions.url).response)
+                response.to_h.must_equal instructions.expected
               end
             end
           end
         end
       end
     end
   end
+
+  class Instructions
+    def initialize(filepath)
+      @filepath = filepath
```

Comment:
I think you may have misunderstood my previous review comment here. Calling this a `filepath` when it's not actually one in lots of cases is much *more* confusing and misleading than what you previously had. A consumer of this class shouldn't need to upgrade a filename to a `Filepath` obvious just to instantiate this (though it should certainly cope with being given one). But having access to one internally is cleaner — i.e. I was suggesting that `new` should accept a `filename`, but there should also be a `filepath` method that upgrades that, and is then used throughout.